csharp/ql/src/semmle/code/csharp/Stmt.qll

```diff
@@ -1254,6 +1285,29 @@ class LocalConstantDeclStmt extends LocalVariableDeclStmt, @const_decl_stmt {
   override string toString() { result = "const ... ...;" }
 }
 
+/**
+ * A `using` declaration statement, for example
+ *
+ * ```
+ * using FileStream f = File.Open("settings.xml");
+ * ```
+ */
+class UsingDeclStmt extends LocalVariableDeclStmt, UsingStmt, @using_decl_stmt {
```

Comment:
Could you please add a test to [Assignables.cs](https://github.com/Semmle/ql/blob/master/csharp/ql/test/library-tests/assignables/Assignables.cs) to check that using declarations are picked up as assignments?

Reply:
I also think we may have to update the queries `cs/local-not-disposed` and `cs/dispose-not-called-on-throw` to account for using declarations.